trio/_file_io/_path.py

```diff
@@ -0,0 +1,138 @@
+
+
+__all__ = ['AsyncPath']
+
+
+def _forward_factory(cls, attr_name, attr):
+    @wraps(attr)
+    def wrapper(self, *args, **kwargs):
+        attr = getattr(self._wrapped, attr_name)
+        value = attr(*args, **kwargs)
+        if isinstance(value, cls._forwards):
+            # re-wrap methods that return new paths
+            value = cls._from_wrapped(value)
+        return value
+
+    return wrapper
+
+
+def thread_wrapper_factory(cls, meth_name):
+    @async_wraps(cls, Path, meth_name)
+    async def wrapper(self, *args, **kwargs):
+        meth = getattr(self._wrapped, meth_name)
+        func = partial(meth, *args, **kwargs)
+        value = await trio.run_in_worker_thread(func)
+        if isinstance(value, cls._wraps):
+            value = cls._from_wrapped(value)
+        return value
+
+    return wrapper
+
+
+class AsyncAutoWrapperType(type):
+    def __init__(cls, name, bases, attrs):
+        super().__init__(name, bases, attrs)
+
+        cls._forward = []
+        type(cls).generate_forwards(cls, attrs)
+        type(cls).generate_wraps(cls, attrs)
+        type(cls).generate_magic(cls, attrs)
+
+    def generate_forwards(cls, attrs):
+        # forward functions of _forwards
+        for attr_name, attr in cls._forwards.__dict__.items():
+            if attr_name.startswith('_') or attr_name in attrs:
+                continue
+
+            if isinstance(attr, property):
+                cls._forward.append(attr_name)
+            elif isinstance(attr, types.FunctionType):
+                wrapper = _forward_factory(cls, attr_name, attr)
+                setattr(cls, attr_name, wrapper)
+            else:
+                raise TypeError(attr_name, type(attr))
+
+    def generate_wraps(cls, attrs):
+        # generate wrappers for functions of _wraps
+        for attr_name, attr in cls._wraps.__dict__.items():
+            if attr_name.startswith('_') or attr_name in attrs:
+                continue
+
+            if isinstance(attr, classmethod):
+                setattr(cls, attr_name, attr)
+            elif isinstance(attr, types.FunctionType):
+                wrapper = thread_wrapper_factory(cls, attr_name)
+                setattr(cls, attr_name, wrapper)
+            else:
+                raise TypeError(attr_name, type(attr))
+
+    def generate_magic(cls, attrs):
+        # generate wrappers for magic
+        for attr_name in cls._forward_magic:
+            attr = getattr(cls._forwards, attr_name)
+            wrapper = _forward_factory(cls, attr_name, attr)
+            setattr(cls, attr_name, wrapper)
+
+
+class AsyncPath(metaclass=AsyncAutoWrapperType):
+    """:class:`trio.AsyncPath` is a :class:`~pathlib.Path` wrapper executes concrete
+    non-computational Path methods in :meth:`trio.run_in_worker_thread`.
+
+    """
+
+    _wraps = Path
+    _forwards = PurePath
+    _forward_magic = [
+        '__str__', '__bytes__',
+        '__eq__', '__lt__', '__le__', '__gt__', '__ge__'
```

Comment:
> actually call them the way we've done this? 

Yeah; I played with it for awhile, and `__str__` at least definitely works.

I'll add tests for the equalities too.

> tp == sp returns True? What about sp == tp?

Should be `True` for both.